enverus-ea/parquetjs
# Read decimal columns

Problem
=======
This package cannot read parquet files with Decimal column types.
========
* Modified the app to pass precision/scale properties from the schema down to the `decodeValues` function in `plain.ts`
* Modified `plain.ts` to use precision/scale for `DECIMAL` types to properly decode decimal columns

Change summary:
---------------
* Tidy, well formulated commit message
* Another great commit message
* Something else I/we did

Steps to Verify:
----------------
1. Download this [parquet file](https://usaz02prismdevmlaas01.blob.core.windows.net/ml-job-config/dataSets/multiple-unsupported-columns.parquet?sv=2020-10-02&st=2023-01-09T15%3A28%3A09Z&se=2025-01-10T15%3A28%3A00Z&sr=b&sp=r&sig=GS0Skk93DCn5CnC64DbnIH2U7JhzHM2nnhq1U%2B2HwPs%3D)
2. Open this parquet with this library `const reader = await parquet.ParquetReader.openFile(<path to parquet file>)`
3. You should be able to load the data 
4. Many rows should have numeric data with decimal places loaded for columns such as `DrillLineDiameterIn`, `FloorHeightFt`, `RotaryTableSizeIn`.
